dbus_service.py

```diff
@@ -598,11 +604,40 @@ def get_values_for_inverter(self):
 
         elif self.dtuvariant == constants.DTUVARIANT_TEMPLATE:
             # logging.debug("JSON data: %s" % meter_data)
-            power = float(get_nested(meter_data, self.custpower) * float(self.custpower_factor))
-            pvyield = float(get_nested(meter_data, self.custtotal) * float(self.custtotal_factor))
-            voltage = float(get_nested(meter_data, self.custvoltage))
-            dc_voltage = float(get_nested(meter_data, self.custdcvoltage))
-            current = float(get_nested(meter_data, self.custcurrent))
+            get_power = get_nested(meter_data, self.custpower)
+            power_value = try_get_value(get_power, float, self.custpower_default)
+            if isinstance(power_value, float) or isinstance(power_value, int):
```

Comment:
(several minutes and thoughts later): Ah this can be configured as "None", or is None by Default. So we get None (or a differently configured default) when the value can not be correctly converted.
Hm. How can we make this more readable?

Maybe another function (with a speaking name) helps here? The function should contain line e.g. 608 to 612.  It will be used several times.

How to call it? Hm. convert_and_multiply_or_default? Or just convert_and_multiply so it could be used like this: `convert_and_multiply(power_value, factor=self.custpower_factor, default=self.custpower_default)`  (type is fixed float here)

Reply:
I will refactor that in a method as soon wen know that this is working as expected.